### PR TITLE
CXP-321: update error management technical stack

### PR DESCRIPTION
**Pull request by pchasle**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ConstraintViolationNormalizer.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * @author Pierre Jolly <pierre.jolly@akeneo.com>
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ConstraintViolationNormalizer extends ViolationNormalizer
+{
+    /**
+     * @param ConstraintViolationInterface $object
+     */
+    public function normalize($object, $format = null, array $context = []): array
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ConstraintViolationNormalizer.php:23]: Please when you put PHPDoc, put the complete one.

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+/**
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductDomainErrorNormalizer implements NormalizerInterface
+{
+    /**
+     * @param IdentifiableDomainErrorInterface $object
+     */
+    public function normalize($object, $format = null, array $context = []): array
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php:22]: idem

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        $data = [
+            'type' => 'domain_error',
+            'domain_error_code' => $object->getErrorIdentifier()
+        ];
+
+        if ($object instanceof \Throwable) {
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php:29]: I wonder if we should add `else { return; }` could we have errors that were not thrown?

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+use Akeneo\Pim\Enrichment\Component\Error\IdentifiableDomainErrorInterface;
+use Akeneo\Pim\Enrichment\Component\Error\TemplatedErrorMessageInterface;
+use Akeneo\Pim\Enrichment\Component\Product\Model\ProductInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * @copyright 2020 Akeneo SAS (http://www.akeneo.com)
+ * @license http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class ProductDomainErrorNormalizer implements NormalizerInterface
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php:17]: I don't see the spec.

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php (hunk 3 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php:29]: If that is the case you won't have the message, but the domain_error_identifier will still be here

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php (hunk 4 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Normalizer/ProductDomainErrorNormalizer.php:17]: Yes, I made the changes on the existent tests to have a green CI, I'm working on adding all the tests for the new classes. Can be done on this PR or another one

**Pull request merged**